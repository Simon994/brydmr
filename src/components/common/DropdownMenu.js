import React, {useRef} from 'react'

import ToggleActive from './ToggleActive'
import menu from '../../styles/assets/menu.png'
import Logo from '../../styles/assets/Logo.png'

export default function DropdownMenu() {
  const dropdownRef = useRef(null);
  const [isActive, setIsActive] = ToggleActive(dropdownRef, false);
  const onClick = () => setIsActive(!isActive);

  return (
    <div className='outer-menu-container'>
      <img 
        src={Logo}
        alt='ByrdMR logo'
        className='byrd-logo'
      />
      <div className='inner-menu-container'>
        <button onClick={onClick} className='menu-trigger'>
          <img
            src={menu}
            alt='Menu button icon'
          />
        </button>
        <nav
          ref={dropdownRef}
          className={`menu ${isActive ? 'active' : 'inactive'}`}
        >
          <ul>
            <li>STORE</li>
            <li>TUTORIALS</li>
            <li>BLOG</li>
            <li>CONTACT</li>
          </ul>
        </nav>
      </div>
    </div>
  );
}

