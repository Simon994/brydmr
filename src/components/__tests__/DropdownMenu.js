import React from 'react'
import renderer from 'react-test-renderer'
import DropdownMenu from '../common/DropdownMenu'


describe('The Menu', () => {
  it('renders as expected', () => {
    const tree = renderer.create(<DropdownMenu/>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
