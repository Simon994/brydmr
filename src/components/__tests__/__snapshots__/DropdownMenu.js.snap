// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Menu renders as expected 1`] = `
<div
  className="outer-menu-container"
>
  <img
    alt="ByrdMR logo"
    className="byrd-logo"
    src="Logo.png"
  />
  <div
    className="inner-menu-container"
  >
    <button
      className="menu-trigger"
      onClick={[Function]}
    >
      <img
        alt="Menu button icon"
        src="menu.png"
      />
    </button>
    <nav
      className="menu inactive"
    >
      <ul>
        <li>
          <a
            href="#"
          >
            STORE
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            exact to=
            /tutorials
            className='nav-link' activeClassName='selected' &gt;
            TUTORIALS
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="#"
          >
            BLOG
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            exact to=
            /contact
            className='nav-link' activeClassName='selected' &gt;
            CONTACT
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
